### DISCUSSION TITLE
Documentation clarification needed for rEFInd part

### DISCUSSION THREAD

**Discussion by thepenguinthatwants** [zbm-dev/zfsbootmenu#337] [Questions]
Hi,
I tried to get zfsbootmenu to work using efibootmgr, but I didn't manage to get it work as it just nagged "BootDevice not found".
So now I've set 
/boot/efi/EFI/ZBM

to contain 
```
/boot/efi/EFI/ZBM/zfsbootmenu.EFI
/boot/efi/EFI/ZBM/refind.conf
```
I've added these lines as my zpool is called zroot. Should there be more stuff in the refind.conf file? I usually use systemd-boot so unsure how the refind.conf should work.
```
"Boot default"  "zbm.prefer=zroot ro quiet loglevel=0 zbm.skip"
"Boot to menu"  "zbm.prefer=zroot ro quiet loglevel=0 zbm.show"
```

Refind just says no eligible ESP's found when I click Install refind to disk from the first restart.

**Comment by zdykstra** [comment]
I'm not clear on what you mean by "click install". The only mechanism to actually set/install `rEFInd` as the EFI boot loader is via the helpe `refind-install`. You'll want EFI vars mounted and your ESP mounted for that.

**Comment by ahesford** [comment]
If you want to directly boot the ZBM image, note that some UEFI implementations (Dell seems to be a common culprit) don't work properly with efibootmgr. I've had success by managing the boot options directly from the Dell setup interface; create a new entry, select zfsbootmenu.EFI from the file chooser and you should be good to go. The firmware interface adds a bunch of device addressing information before the executable path; you can see this information by looking at the output of efibootmgr after adding an entry from the firmware setup.
